beekeeper-path-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/handler/ExpiredMessageHandler.java

```diff
@@ -0,0 +1,102 @@
+
+import com.expediagroup.beekeeper.core.model.EntityHousekeepingTable;
+import com.expediagroup.beekeeper.core.model.LifecycleEventType;
+import com.expediagroup.beekeeper.scheduler.apiary.filter.EventTypeListenerEventFilter;
+import com.expediagroup.beekeeper.scheduler.apiary.filter.ListenerEventFilter;
+import com.expediagroup.beekeeper.scheduler.apiary.filter.TableParameterListenerEventFilter;
+import com.expediagroup.beekeeper.scheduler.apiary.filter.WhitelistedListenerEventFilter;
+import com.expediagroup.beekeeper.scheduler.apiary.model.ExpiredEventModel;
+
+@Component
+public class ExpiredMessageHandler extends MessageEventHandler<EntityHousekeepingTable, ExpiredEventModel> {
+
+  private static final Logger log = LoggerFactory.getLogger(ExpiredMessageHandler.class);
+  private static final String EXPIRED_DATA_RETENTION_PERIOD_PROPERTY_KEY = "beekeeper.expired.data.retention.period";
+  private static final LifecycleEventType LIFECYCLE_EVENT_TYPE = LifecycleEventType.EXPIRED;
+  private static final List<Class<? extends ListenerEvent>> EVENT_CLASSES = List.of(AlterTableEvent.class);
+
+  private final List<ListenerEventFilter> filters;
+
+  @Autowired
+  public ExpiredMessageHandler(
+      @Value("${properties.beekeeper.default-expiration-delay}") String cleanupDelay
+  ) {
+    super(cleanupDelay, EXPIRED_DATA_RETENTION_PERIOD_PROPERTY_KEY, LIFECYCLE_EVENT_TYPE);
+    this.filters = List.of(
+        new EventTypeListenerEventFilter(EVENT_CLASSES),
+        new TableParameterListenerEventFilter(),
+        new WhitelistedListenerEventFilter());
+  }
+
+  public ExpiredMessageHandler(
+      @Value("${properties.beekeeper.default-expiration-delay}") String cleanupDelay,
+      List<ListenerEventFilter> filters
+  ) {
+    super(cleanupDelay, EXPIRED_DATA_RETENTION_PERIOD_PROPERTY_KEY, LIFECYCLE_EVENT_TYPE);
+    this.filters = filters;
+  }
+
+  @Override
+  protected List<ListenerEventFilter> getFilters() {
+    return filters;
+  }
+
+  @Override
+  protected EntityHousekeepingTable generateHousekeepingEntity(ExpiredEventModel eventModel,
+      ListenerEvent listenerEvent) {
+    EntityHousekeepingTable.Builder builder = new EntityHousekeepingTable.Builder()
+        .housekeepingStatus(eventModel.getHousekeepingStatus())
+        .creationTimestamp(LocalDateTime.now())
+        .cleanupDelay(extractCleanupDelay(listenerEvent))
+        .lifecycleType(eventModel.getLifecycleEventType().name())
+        .clientId(CLIENT_ID)
+        .tableName(listenerEvent.getTableName())
+        .databaseName(listenerEvent.getDbName());
+
+    return builder.build();
+  }
+
+  @Override
+  protected List<ExpiredEventModel> generateEventModels(ListenerEvent event) {
+    List<ExpiredEventModel> eventPaths = new ArrayList<>();
+
+    switch (event.getEventType()) {
+    case ALTER_TABLE:
+      eventPaths.add(new ExpiredEventModel(LIFECYCLE_EVENT_TYPE, SCHEDULED));
+      break;
```

Comment:
We could have a default setting here that throws an exception as we shouldn't expect to reach it

Reply:
Well, all the other events will throw that exception then right? Shouldn't we just ignore events we don't need and only care about the ones we want?

Reply:
Hmm, I think for the other flow we decided that if an event lands here which doesn't match one of the cases, something has gone wrong i.e. the filters should have filtered it out. So if we want only alter tables to get to this point, everything else will have to be filtered somehow.

Reply:
Ah yeah you're right. Sure I'll do that :-)